### ISSUE TITLE
Double-GZip when using module with connect/express.compress()

### ISSUE THREAD

**Issue by chakrit** [dustyburwell/connect-file-cache#7]
TL;DR: Detect already-gzipped content stream or remove compression feature altogether now that there is `connect.compress()` available.

---

I'm using this module through `connect-assets`. I don't have a working test case for this yet but please bear with me as I've confirmed this to be reproducible and can point you to problemetic lines if you want.

I'm using `connect-assets` in conjunction with `connect.compress()` The problem is that `connect.compress()` will compresses all compatible content **wether or not it is already compressed**. So if me or another user, configures express or connect like this:

```
app.use(express.compress())
app.use(require('connect-assets')())
```

This will results in all assets files response being double-gzipped. (because connect-file-cache will write already gzipped-data from its cache to the response which is already piped through the gzip stream setup by `express.compress()`.

See here: https://github.com/TrevorBurnham/connect-file-cache/blob/master/lib/cache.js#L99

I have to do something like this: `curl 0.0.0.0:7000/js/site.js -H "Accept-Encoding: gzip" | gunzip | gunzip` for the content to render.

Although, this can be worked around by switching the `app.use` order. I believe this should be fixed because more people will now use the vanilla `express.compress()` call and the order of middleware placement really shouldn't have effects on how files are being compressed or not.

This can get confusing because it will often shows up when people are re-arranging the middlewares `app.use` calls and suddenly getting a bad response from all the asets without an idea of what's going on.

**Comment by rprieto** [comment]
That would be great!
As a workaround, my current setup is:

```
app.use(connectAssets({src: path.join(__dirname, 'assets')}));
// Careful, add gzip compression *after* connectAssets, since it already GZIPs its responses
app.use(connect.compress());
app.use(app.router);
app.use(express.static(path.join(__dirname, 'assets')));
```

Which means:
- `connect-assets` can serve (and gzip) any assets it finds
  subsequent middlewares don't get executed
- other requests such as custom routes and static content gets served further down the request pipeline
  they get compressed by `connect-compress` on the way

**Comment by chakrit** [comment]
@rprieto nice. thanks!

Closing since this: https://github.com/senchalabs/connect/pull/659 is now available for the latest connect.

So the fix is: update express and/or connect.
